### ISSUE TITLE
[Profiler] Device duration of all operators is zero in tensorboard

### ISSUE THREAD

**Issue by OceanF0rever** [pytorch/torchtune#1662]
I run fine-tune Llama3-8B-Instruct with LoRA on a single device using torchtune, and enable pytorch profiler.
I'm not sure why the device duration for all operators is displayed as zero in TensorBoard.
<img width="460" alt="config" src="https://github.com/user-attachments/assets/fb44c1fc-1823-4b7c-96eb-e0b601c690d0">
<img width="931" alt="operators" src="https://github.com/user-attachments/assets/f824a6b4-6b4f-449e-ba0f-2cbd02f880f6">
<img width="934" alt="kernels" src="https://github.com/user-attachments/assets/55721fc6-32f4-41c2-9ba1-de9c4d0e07a1">

**Comment by felipemello1** [comment]
Hey @OceanF0rever , thanks for raising the issue! Do you have any intuition about what could be the root of the issue?

@janeyx99 , have you seen this before in your traces?

**Comment by janeyx99** [comment]
I haven't seen the Tensorboard UI, but it looks like the top one is recording the dispatcher overhead of the ATen ops, and the actual kernels do compute the time correctly. I would expect the kernel numbers to cascade up though, cc @aaronenyeshi for how to interpret tensorboard

**Comment by OceanF0rever** [comment]
> Hey @OceanF0rever , thanks for raising the issue! Do you have any intuition about what could be the root of the issue?
> 
> @janeyx99 , have you seen this before in your traces?

<img width="781" alt="atenmm" src="https://github.com/user-attachments/assets/aa318678-0727-45f2-ab94-5b4f09215e29">
<img width="782" alt="kernel" src="https://github.com/user-attachments/assets/2f3bcc74-d2f1-4359-b9bc-04292cb44a20">

I checked the pt.trace.json file, as shown in the screenshot. The category of aten::mm is cpu_op, with only one duration, and it is not associated with any CUDA kernel. The category of the CUDA kernel is kernel, and the duration should correspond to the device duration.
Is there a specific configuration option that needs to be enabled to associate aten::op with CUDA kernels?

**Comment by aaronenyeshi** [comment]
I agree with @janeyx99 , the top image is showing the Operator view, which shows host side ops that do not use the device, and the bottom image is showing the GPU kernels that will use the device. cc @sraikund16 

> Is there a specific configuration option that needs to be enabled to associate aten::op with CUDA kernels?
There should be a flow event (ph = f and ph = s), which corresponds host ops with CUDA kernels. The s event is the start of the flow, and the f event is the end of a flow. Usually starts from the host event, and ends on the kernel event. Can you opne the file directly in url: `chrome://tracing` on your chrome browser?

**Comment by OceanF0rever** [comment]
> I agree with @janeyx99 , the top image is showing the Operator view, which shows host side ops that do not use the device, and the bottom image is showing the GPU kernels that will use the device. cc @sraikund16
> 
> > Is there a specific configuration option that needs to be enabled to associate aten::op with CUDA kernels?
> 
> There should be a flow event (ph = f and ph = s), which corresponds host ops with CUDA kernels. The s event is the start of the flow, and the f event is the end of a flow. Usually starts from the host event, and ends on the kernel event. Can you opne the file directly in url: `chrome://tracing` on your chrome browser?

The trace view is fine, whether using TensorBoard or directly opening the JSON file with chrome://tracing. I think that Torchtune's default configuration when using PyTorch Profiler is causing this issue. In the PyTorch tutorials,  https://pytorch.org/tutorials/_static/img/profiler_operator_view.png，you can see in this image that the operators have both host duration and device duration, and PyTorch Profiler can associate aten::op with the kernel.

**Comment by OceanF0rever** [comment]
https://github.com/pytorch/pytorch/issues/82712
This issue seems to be a bug in the PyTorch profiler itself. It was fixed in this year, but I am using the 2.4.0 release version, and the problem still persists.
